lua/SS13.lua

```diff
@@ -28,21 +26,20 @@ function SS13.await(thing_to_call, proc_to_call, ...)
 	if thing_to_call == SS13.global_proc then
 		proc_to_call = "/proc/" .. proc_to_call
 	end
-	local args = table.pack(thing_to_call, proc_to_call, ...)
-	args["n"] = nil
-	local promise = SS13.new("/datum/promise", table.unpack(args))
+	local promise = SS13.new("/datum/auxtools_promise", thing_to_call, proc_to_call, ...)
 	while promise:get_var("status") == 0 do
 		sleep()
 	end
 	return promise:get_var("return_value"), promise:get_var("runtime_message")
 end
 
-function SS13.wait(time, _timer)
+function SS13.wait(time, timer)
 	local index = #__yield_table + 1
 	local callback = SS13.new("/datum/callback", SS13.SSlua, "queue_resume", SS13.state, index)
-	local timer = dm.global_proc("_addtimer", callback, time*10, 8, _timer, "lua/SS13.lua", 43)
+	local source, line = debug.info(1, "sl")
+	local timedevent = dm.global_proc("_addtimer", callback, time*10, 8, timer, source, line+1)
```

Comment:
```suggestion
	local timedevent = dm.global_proc("_addtimer", callback, time*10, 8, timer, debug.info(1, "sl"))
```

Lua will pass through both, so long as more stuff isn't added to the end.